src/runtime/typemanager.cs

```diff
@@ -343,30 +388,76 @@ internal static IntPtr CreateMetaType(Type impl)
             // We need space for 3 PyMethodDef structs, each of them
             // 4 int-ptrs in size.
             IntPtr mdef = Runtime.PyMem_Malloc(3 * 4 * IntPtr.Size);
+            Debug.Assert(4 * IntPtr.Size == Marshal.SizeOf(typeof(PyMethodDef)));
             IntPtr mdefStart = mdef;
+            ThunkInfo thunk = Interop.GetThunk(typeof(MetaType).GetMethod("__instancecheck__"), "BinaryFunc");
+            slotsHolder.KeeapAlive(thunk.Target);
+
+            {
+                IntPtr mdefAddr = mdef;
+                slotsHolder.AddDealloctor(() =>
+                {
+                    IntPtr t = type;
+                    IntPtr tp_dict = Marshal.ReadIntPtr(t, TypeOffset.tp_dict);
+                    if (Runtime.PyDict_DelItemString(tp_dict, "__instancecheck__") != 0)
+                    {
+                        Runtime.PyErr_Print();
+                    }
+                    FreeMethodDef(mdefAddr);
```

Comment:
Why do we open a scope and create a new local variable that just copies a local variable that already exists?

Reply:
Oh, I get it -- so that the lambda grabs the right value.

Feels like we could refactor these two scopes into a function, so it's a bit less mysterious.